nixos/modules/programs/qgroundcontrol.nix

```
@@ -0,0 +1,39 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.qgroundcontrol;
+in
+{
+
+  options = {
+    programs.qgroundcontrol = {
+      enable = lib.mkEnableOption "qgroundcontrol";
+
+      package = lib.mkPackageOption pkgs "qgroundcontrol" {};
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    # ModemManager is known to interfere with serial connections;
+    # qgroundcontrol recommends disabling it
+    systemd.services.ModemManager.enable = lib.mkDefault false;
+
+    # Security wrapper
+    security.wrappers.qgroundcontrol = {
+      source = "${pkgs.qgroundcontrol}/bin/qgroundcontrol";
```

This binary name is off. The package seems to provide `QGroundControl`. Maybe it's better to just retrieve it through a function anyway.

```suggestion
      source = lib.getExe pkgs.qgroundcontrol;
```